### PR TITLE
Playback speed is not displayed correctly in video player #40646

### PR DESCRIPTION
**Pull request by superdev100**
### Fixed Issues

$ https://github.com/Expensify/App/issues/40646
PROPOSAL: https://github.com/Expensify/App/issues/40646#issuecomment-2067790768

### Tests

For MacOS: Chrome / Safari & MacOS: Desktop
1. Upload two videos(`Video1`, `Video2`).
2. Play `Video1`.
3. Play `Video2`.
4. Change playback speed of `Video2` to 2.
5. Then change playback speed of `Video1` to 0.25 (or any other)
6. Check if playback speed of `Video2` is 2.
7. Play `Video1` and check if speed is 0.25.

Other platforms
1. Upload two videos(`Video1`, `Video2`).
2. Play `Video1` and change speed to 0.25(or any other)
3. Play `Video2` and check that speed is 1.

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/161328186/2bdca3a8-9426-456a-ae85-ac7ab9b0ae5a

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/161328186/5d4900f3-cfbf-4cc1-ba29-37ec743e12aa

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/161328186/c3fea42b-0004-49ee-8e2d-364809e66ee0

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/161328186/c7f81c72-ac94-41b7-bbaf-ab82f355d99c

</details>

### CODE REVIEW

File: src/components/VideoPlayerContexts/VideoPopoverMenuContext.tsx [hunk 1]
```diff
@@ -18,13 +19,14 @@ function VideoPopoverMenuContextProvider({children}: ChildrenProps) {
     const [currentPlaybackSpeed, setCurrentPlaybackSpeed] = useState<PlaybackSpeed>(CONST.VIDEO_PLAYER.PLAYBACK_SPEEDS[2]);
     const {isOffline} = useNetwork();
     const isLocalFile = currentlyPlayingURL && CONST.ATTACHMENT_LOCAL_URL_PREFIX.some((prefix) => currentlyPlayingURL.startsWith(prefix));
+    const playerRef = useRef<VideoWithOnFullScreenUpdate | null>(null);
```
Comment (hungvu193) [inline src/components/VideoPlayerContexts/VideoPopoverMenuContext.tsx:22]: Can we give it a more meaningful name? IE: videoPopoverMenuRef
```suggestion
    const videoPopoverMenuPlayerRef = useRef<VideoWithOnFullScreenUpdate | null>(null);
```

**Review by hungvu193 — CHANGES_REQUESTED** [review]

**Comment by hungvu193** [comment]
@KMichel1030 I recommend that you should only mark PR ready when you've completed all the video and checklist.

**Comment by hungvu193** [comment]
Any update here please? @KMichel1030

**Comment by hungvu193** [comment]
We have conflicts here @KMichel1030

**Comment by KMichel1030** [comment]
@hungvu193 
Sorry for late reply.
The conflict has occurred because of [this issue](https://github.com/Expensify/App/pull/41054).
I will resolve conflict asap.

**Comment by hungvu193** [comment]
Is there any chance that you can complete the PR author checklist today?

**Comment by KMichel1030** [comment]
@hungvu193 
Sorry for delay.
Currently, I have problem with my dev env. I tried to build other platforms but it's difficult to find problem.
I'm still working to build other platforms.
But I'm wondering if there's any other way we can proceed. Actually, I can't sure how long it will take.
Thank you for your understanding.

**Comment by hungvu193** [comment]
Did you try npm run clean?

**Comment by KMichel1030** [comment]
Sorry, I've tried `npm run clean` but still have [this error](https://expensify.slack.com/archives/C01GTK53T8Q/p1714900762191449).

**Comment by hungvu193** [comment]
can you please post the full logs? Also if you can build ios, can you build android? I only saw you complete the record for desktop and mac/chrome

**Comment by KMichel1030** [comment]
@hungvu193 
I've run web & desktop versions but I have trouble with other versions.
Regarding android, I haven't built it even once.
I'm sorry but you know, I'm a new contributor and this is first time to work on other platforms.
I'm trying hard but I'm wondering if we can proceed with your check. Of course, only if it is not violation of contributing rule.
cc: @blimpich

**Comment by KMichel1030** [comment]
Of course, I've followed guide. I built ios version once, but then error persists.
I'm trying for over a week but can't find solution. I have problem on ios so I can't afford to build other versions.

**Comment by hungvu193** [comment]
Please post the full logs to expensify open source so we can help you. I'll verify the changes in the mean time.

**Comment by hungvu193** [comment]
Can you please merge with main? @KMichel1030

**Comment by hungvu193** [comment]
Please see the video, I don't think it's working for IOS. I got a maximum update also.

https://github.com/Expensify/App/assets/16502320/dbefde4c-0ae2-44df-a814-e69417ec1f4a

**Comment by KMichel1030** [comment]
Ok, I will check. Thank you.

File: src/components/VideoPlayer/BaseVideoPlayer.tsx [hunk 2]
```diff
@@ -94,8 +95,14 @@ function BaseVideoPlayer({
...
     const showPopoverMenu = (event?: GestureResponderEvent | KeyboardEvent) => {
-        setIsPopoverVisible(true);
         videoPopoverMenuPlayerRef.current = videoPlayerRef.current;
+        videoPlayerRef.current?.getStatusAsync().then((status) => {
+            if (!('rate' in status && status.rate)) {
+                return;
+            }
+            setIsPopoverVisible(true);
+            setCurrentPlaybackSpeed(status.rate as PlaybackSpeed);
+        });
```
Comment (KMichel1030) [inline src/components/VideoPlayer/BaseVideoPlayer.tsx:105]: @hungvu193 

> Please see the video, I don't think it's working for IOS.

This bug maybe related with delay of this function.  Run time of this function is affected by some factors like video size, quality and network status.

Could you please check changes?

> I got an infinite loop render also

I can't sure about the infinite loop render you mentioned. Could you please explain in more detail?

**Comment by hungvu193** [comment]
Still doesn't work for me, can you build IOS btw?

**Comment by KMichel1030** [comment]
Now I'm upgrading mac vmware and xcode.
I will build ios by next Monday at worst case.
Thank you for your understanding.

**Comment by hungvu193** [comment]
Ok, thanks for looking into it.

File: src/components/VideoPlayer/BaseVideoPlayer.tsx (hunk 2 above)
Comment (hungvu193) [reply src/components/VideoPlayer/BaseVideoPlayer.tsx:105]: I got the loop few times so I can not give you a consistent reproducible steps but let me try.

**Comment by KMichel1030** [comment]
Could you please let me know if it is working on other platforms?
Doesn't work on only ios?

**Comment by hungvu193** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by hungvu193** [comment]
> Could you please let me know if it is working on other platforms? Doesn't work on only ios? It will be helpful if you post test video for me. I can't sure what the problem is.

The problem will be with small screen, since we will use attachment for them to display (wrap) the video player instead of using VideoPlayer itself directly.

**Comment by hungvu193** [comment]
Friendly bump here @KMichel1030

**Comment by KMichel1030** [comment]
@hungvu193 
I'm very sorry but I still working on it. I will make it run in a few days.

> The problem will be with small screen, since we will use attachment modal for them to display (wrap) the video player instead of using VideoPlayer itself directly.

I've tested it with smartphone size in web and it's working properly. So it seems it's not problem with small screen.
Could you please test again and let me know test steps for error case? Thank you for understanding.

https://github.com/Expensify/App/assets/161328186/fd6c1460-b0a0-4011-840c-3f4c41f9d41a

**Comment by hungvu193** [comment]
Cool. Thanks for letting me know. Can you please merge main again? I'll take another few tests

**Comment by KMichel1030** [comment]
@hungvu193 
Thank you. I've merged main branch.

**Comment by KMichel1030** [comment]
Hi, @hungvu193 
I've checked on ios and confirmed it's working properly.

**Comment by hungvu193** [comment]
Cool, please complete the author checklist while I'm checking it rn

**Comment by hungvu193** [comment]
Feels like server is down, I can't upload the video, I'll take another test later today

Screenshot 2024-05-15 at 15 08 13

**Comment by KMichel1030** [comment]
@hungvu193 
Thank you.
BTW, I try to test on ios safari but so this error. Could you please let me know the solution? I tried several solutions but failed.
image

**Comment by hungvu193** [comment]
Make sure you setup https: https://github.com/Expensify/App?tab=readme-ov-file#configuring-https
Also can you try with `https://127.0.0.1:8082` instead?

**Comment by KMichel1030** [comment]
@hungvu193 
Thank you, I could open page with `https://127.0.0.1:8082`. I will build android version asap.

**Comment by hungvu193** [comment]
Can you merge main again? MenuItem is broken on our branch

**Comment by KMichel1030** [comment]
@hungvu193 
I've merged main branch. But I can't sure that the problem is.
Could you please tell me more about MenuItem crash?

**Comment by hungvu193** [comment]
> @hungvu193 I've merged main branch. But I can't sure that the problem is. Could you please tell me more about MenuItem crash?

No worries, it's not related to this PR

**Comment by hungvu193** [comment]
So I got this one on iOS, open any video, change the speed, close the modal and open that video again, now the speed is reset to 1. But I think that's fine. I didn't find any other problem so far. I'll complete checklist in a while

**Comment by hungvu193** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



https://github.com/Expensify/App/assets/16502320/61e07231-c281-427f-a463-dd82cd79ad11


</details>

<details>
<summary>Android: mWeb Chrome</summary>



https://github.com/Expensify/App/assets/16502320/316dff58-483c-4a68-92b3-e941b00fa5cb


</details>

<details>
<summary>iOS: Native</summary>



https://github.com/Expensify/App/assets/16502320/57383b2c-8505-4ec0-89f5-fe4cdadeedcb


</details>

<details>
<summary>iOS: mWeb Safari</summary>


https://github.com/Expensify/App/assets/16502320/17f083cb-216c-4fb6-9c6d-582eb4ea7ff7



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>





https://github.com/Expensify/App/assets/16502320/bb79b83f-dd6a-4d35-889c-f47475da9429



</details>

<details>
<summary>MacOS: Desktop</summary>



https://github.com/Expensify/App/assets/16502320/5ea7a8fa-df67-4110-844b-06ef02097679

</details>

**Review by hungvu193 — APPROVED** [review]
LGTM!

**Review by blimpich — APPROVED** [review]
LGTM

File: src/components/VideoPlayerContexts/VideoPopoverMenuContext.tsx [hunk 3]
```diff
@@ -24,9 +24,9 @@ function VideoPopoverMenuContextProvider({children}: ChildrenProps) {
     const updatePlaybackSpeed = useCallback(
         (speed: PlaybackSpeed) => {
             setCurrentPlaybackSpeed(speed);
-            currentVideoPlayerRef.current?.setStatusAsync?.({rate: speed});
+            videoPopoverMenuPlayerRef.current?.setStatusAsync?.({rate: speed});
```
Comment (situchan) [inline src/components/VideoPlayerContexts/VideoPopoverMenuContext.tsx:27]: There was [regression](https://github.com/Expensify/App/issues/42519) from this change.
More details: https://github.com/Expensify/App/issues/42519#issuecomment-2143431227

**Pull request merged**